packages/web3-shared/src/hooks/useMulticall.ts

```diff
@@ -64,10 +54,12 @@ export function useMulticallCallback() {
                     results: returnData,
                 })
             } catch (error) {
-                setMulticallState({
-                    type: MulticalStateType.FAILED,
-                    error,
-                })
+                if (error instanceof Error) {
```

Comment:
This change actually changes the semantics and might break the code. Please always re-throw the error